src/main/java/org/example/homework10/task2/Task2.java

```diff
@@ -0,0 +1,75 @@
+package org.example.homework10.task2;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+public class Task2 {
+    private static final DateTimeFormatter PATTERN = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+
+    public static void main(String[] args) {
+
+        Employee e1 = new Employee("Minnie", "Mouse", "01/02/2015");
+        Employee e2 = new Employee("Mickie", "Mouse", "05/08/2000");
+        Employee e3 = new Employee("Daffy", "Duck", "11/02/2011");
+        Employee e4 = new Employee("Daisy", "Duck", "05/03/2013");
+        Employee e5 = new Employee("Goofy", "Dog", "23/07/2020");
+
+        List<Employee> list = new ArrayList<>(Arrays.asList(e1, e2, e3, e4, e5));
+
+        printOrderedList(list, "name");
+        System.out.println();
+        printOrderedList(list, "year");
+    }
+
+    private static void printOrderedList(List<Employee> list, String sortBy) {
+        class CustomEmployee {
+            private String fullName;
+            private Integer yearsWorked;
+            private Employee employee;
+
+            CustomEmployee(Employee employee) {
+                this.employee = employee;
+                fullName = employee.firstName() + ' ' + employee.lastName();
+                yearsWorked = LocalDate.now().getYear() - LocalDate.parse(employee.hireDate(), PATTERN).getYear();
+            }
+
+            @Override
+            public String toString() {
+                return "CustomEmployee {" +
+                        "fullName = '" + fullName + '\'' +
+                        ", yearsWorked = " + yearsWorked +
+                        '}';
+            }
+        }
+
+        List<CustomEmployee> customEmployees = new ArrayList<>(list.size());
+        for (Employee employee : list) {
+            customEmployees.add(new CustomEmployee(employee));
+        }
+
+        Comparator<CustomEmployee> customEmployeeComparator;
```

Comment:
all this stuff could be done a bit easier
```suggestion
                var comparator = new Comparator<CustomEmployee>() {

            @Override
            public int compare(CustomEmployee o1, CustomEmployee o2) {

                if (sortField.equals("name")) {
                    return o1.fullName.compareTo(o2.fullName);
                }
                return o1.yearsWorked - o2.yearsWorked;
            }
        };
```

and after it you can just use it in `list.sort(comparator);`

Reply:
I changed 2 comparators in switch to a single one with if branches.